contrib/cryptomb/private_key_providers/source/cryptomb_private_key_provider.cc

```diff
@@ -29,6 +29,54 @@ void CryptoMbContext::scheduleCallback(enum RequestStatus status) {
   schedulable_->scheduleCallbackNextIteration();
 }
 
+bool CryptoMbEcdsaContext::ecdsaInit(const uint8_t* in, size_t in_len) {
+  if (ec_key_ == nullptr) {
+    return false;
+  }
+
+  const EC_GROUP* group = EC_KEY_get0_group(ec_key_.get());
+  priv_key_ = EC_KEY_get0_private_key(ec_key_.get());
+  if (group == nullptr || priv_key_ == nullptr) {
+    return false;
+  }
+
+  const BIGNUM* order = EC_GROUP_get0_order(group);
+  if (order == nullptr) {
+    return false;
+  }
+
+  // Create an ephemeral key.
+  ctx_ = bssl::UniquePtr<BN_CTX>(BN_CTX_new());
+  if (ctx_ == nullptr) {
+    return false;
+  }
+  BN_CTX_start(ctx_.get());
+  k_ = BN_CTX_get(ctx_.get());
+  do {
+    if (!BN_rand_range(k_, order)) {
+      return false;
+    }
+  } while (BN_is_zero(k_));
+
+  // Extent with zero paddings as CryptoMB expects in_buf_ being sign length.
+  int len = BN_num_bits(order);
+  size_t buf_len = (len + 7) / 8;
+  if (8 * in_len < static_cast<unsigned long>(len)) {
+    in_buf_ = std::make_unique<uint8_t[]>(buf_len);
+    memcpy(in_buf_.get() + buf_len - in_len, in, in_len); // NOLINT(safe-memcpy)
+  } else {
+    in_buf_ = std::make_unique<uint8_t[]>(in_len);
+    memcpy(in_buf_.get(), in, in_len); // NOLINT(safe-memcpy)
+  }
+
+  sig_len_ = ECDSA_size(ec_key_.get());
+  if (sig_len_ > MAX_SIGNATURE_SIZE) {
```

Comment:
One question: Do the RSA and ECDSA have the same max signature size? I saw the ECDSA signature size is twice large than the key, but RSA's signature size is same with the key. Is there any chance the max ECDSA signature large than the RSA one?

Reply:
At least it will not happen in ECDSA P256.

Reply:
ok, got it.